rstudio/rstudio
# Fix issue preventing screen reader mode from enabling

### Intent

Addresses #13710

### Approach

Due to other changes, the previous approach for checking the state of the screen reader flag at startup was no longer getting the current value (I believe it was getting the default value). This code was responsible for deciding if it should announce "screen reader not enabled" and for setting the menu text. Code that runs later was getting the correct value, hence the mismatches.

There was a second symptom of this problem, also related to accessibility. The "tab key moves focus" toggle (Help / Accessibility / Focus / Tab Key Always Moves Focus) was also not showing the correct state (via a checkmark).

Fixed by moving that startup code to the end of `onSessionInit` at which point preferences are happy.

### Automated Tests

None that I'm aware of; it would be nice to have a test to check that this screen reader flag is working, but not sure we have the capability to test something that requires a reload of the IDE.

### QA Notes

Test that the menu text both in Help / Accessibility and in the Command Palette now correctly show the current state of the screen reader toggle.

Also test with a screen reader running: when the screen reader flag is off, you should hear an announcement at startup (often after the Console is read) warning that screen reader mode isn't enabled. Once you enable it and restart you should no longer hear that announcement.

Finally, spot-check that the Tab Key Always Moves Focus command is showing the correct state in the menu Help / Accessibility / Focus : Tab Key Always Moves Focus. The text never changes but there should be a checkmark when it is enabled.

### Documentation

None

### Checklist
